src/components/button/button.tsx

```diff
@@ -1,26 +1,58 @@
 import { Component, Element, h, Host, Prop } from '@stencil/core';
 
+/**
+ * A Material Design button component.
+ *
+ * To render a clear Material button on any html page :
+ * @example
+ * <materials-button>Click me</materials-button>
+ *
+ * @description
+ * You can use any icon from the material-icons collection (https://material.io/tools/icons)
+ *
+ * Code is available on github: https://github.com/GMV-centravet/materials/tree/master/src/components/button
+ *
+ * Material Design guidelines: https://material.io/design/components/buttons.html
+ */
 @Component({
   tag: 'materials-button',
   styleUrl: 'button.scss',
   shadow: true
 })
 export class Button {
 
-  @Prop() icon: string;
-  @Prop() color: 'primary' | 'accent' | 'secondary' | 'error' | string = 'primary';
-  @Prop() type: 'button' | 'submit' | 'reset';
-  @Prop() disabled = false;
-  /**
-   * Attention la valeur du border-radius sera toujours en px !
+  /** The button color, it can be either :
+   *  - a predifined value : 'primary', 'secondary', 'error'.
+   *  - an hexa color code : #225566, #CCC.
+   *  - a css named color : red, blue.
    */
-  @Prop() radius: number | string;
-  @Prop() raised: boolean;
-  @Prop() unelevated: boolean;
-  @Prop() outlined: boolean;
-  @Prop() dense: boolean;
-  @Prop() block: boolean;
-  @Element() btnEl: HTMLButtonElement;
+  @Prop() color: 'primary' | 'secondary' | 'error' | string = 'primary';
+
+  /** A material icon name */
+  @Prop() icon?: string = '';
+
+  /** The button type */
+  @Prop() type: 'button' | 'submit' | 'reset' = 'button';
+
+  /** Disable the button */
+  @Prop() disabled = false;
+
+  /** Render a raised Material button */
+  @Prop() raised: boolean = false;
+
```

Comment:
tslint must raise a warning here because assignation and typing for boolean is redundant

Reply:
We should update tslint rules, we need this for default value in doc generation.